app/models/course/notification.rb

```diff
@@ -0,0 +1,20 @@
+class Course::Notification < ActiveRecord::Base
+  include ApplicationNotificationsHelper
+
+  belongs_to :activity, inverse_of: :course_notifications
+  belongs_to :course, inverse_of: :notifications
+  after_create :send_email
+
+  enum notification_type: { feed: 0, email: 1 }
+
+  private
+
+  # Send out an email immediately if the email column of an activity is true
+  def send_email
+    return unless notification_type == 'email'
+    view_path = notification_view_path(self)
+    course.users.each do |user|
+      ActivityMailer.email(user, activity.object, view_path).deliver_now
```

Comment:
hmm... what this means? If use `deliver_now` to send out emails, it has already been a job and will be sent asynchronously